cmd/s3-gw/app.go

```diff
@@ -86,17 +88,29 @@ type (
 )
 
 func newApp(ctx context.Context, log *Logger, v *viper.Viper) *App {
-	conns, key := getPool(ctx, log.logger, v)
+	conns, key, poolStat := getPool(ctx, log.logger, v)
+
+	signer := user.NewAutoIDSignerRFC6979(key.PrivateKey)
+
+	// authmate doesn't require anonKey for work, but let's create random one.
+	anonKey, err := keys.NewPrivateKey()
+	if err != nil {
+		log.logger.Fatal("newApp: couldn't generate random key", zap.Error(err))
+	}
+	anonSigner := user.NewAutoIDSignerRFC6979(anonKey.PrivateKey)
```

Comment:
...is related to this key? I thought they're supposed to be the same.

Reply:
This would make things more predictable, I think. Consider some logs, for example, you can have one key logged in one place and another in another, then try to make some sense of it.

Reply:
Updated